notebooks/01-Summary-statistics-and-plots.ipynb

```diff
@@ -0,0 +1,313 @@
+    "Circular grains are excluded since their end-to-end length is 0.0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cf4d92f0-b624-452c-9f51-52fe70dfde88",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "df[df[\"circular\"] == False][\"end_to_end_distance\"].plot.hist(figsize=(16,9),\n",
+    "                                                             bins=20,\n",
+    "                                                             title=\"End to End Distance\",\n",
+    "                                                             alpha=0.5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94270659-cc2a-4606-9fa9-80ed5fac82d2",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "### Multiple Images\n",
+    "\n",
+    "Often you will have processed multiple images and you will want to plot the distributions of metrics for each image\n",
+    "separately.\n",
+    "\n",
+    "For this example we duplicate the data and append it, adjusting the values slightly"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "529aedb3-2d4e-4703-86a0-bcba99caee57",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "def scale_df(df: pd.DataFrame, scale:float, image:str) -> pd.DataFrame:\n",
+    "    \"\"\"Scale the numerical values of a data frame. Retains string variables and the index.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    df: pd.DataFrame\n",
+    "        Pandas Dataframe to scale.\n",
+    "    scale: float\n",
+    "        Factor by which to scale the data.\n",
+    "    image: str\n",
+    "        Name for new (dummy) image.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    pd.DataFrame\n",
+    "        Scaled data frame\n",
+    "    \"\"\"\n",
+    "    _df = df[df.select_dtypes(include=['number']).columns] * scale\n",
+    "    _df.reset_index(inplace=True)\n",
+    "    _df[\"image\"] = image\n",
+    "    _df = pd.concat([_df, df[[\"circular\", \"basename\"]]], axis=1)\n",
+    "    _df.set_index([\"image\", \"threshold\", \"molecule_number\"], inplace=True)\n",
+    "    return _df\n",
+    "smaller = scale_df(df, scale=0.9, image=\"smaller\")\n",
+    "larger = scale_df(df, scale=1.25, image=\"larger\")\n",
+    "df_three_images = pd.concat([smaller, df, larger])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6a57eb1a-3093-4ca6-8411-e62ea2614730",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "### Contour Length from Three Processed Images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9820d259-680e-49fd-a1eb-e14d200a0b7a",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "df_three_images[\"contour_lengths\"].groupby(\"image\").plot.hist(figsize=(16,9),\n",
+    "                                bins=20,\n",
```

Comment:
Since different files have different data ranges, having something like `bins=20` would lead to bins that don't match with each other. Would it make sense to add a (possibly commented out) line that says, eg `bins=np.linspace(0, 3e-7, 20)`, so that the user knows how to make plots with aligned bins? 
(The numpy module would need to be imported at the start)

Reply:
My understanding of `bins` is that if its an integer then the plot has that number of equally spaced bins across the range. `np.linspace(start, stop, num=x)` essentially does the same thing.

Its a tricky balance, I appreciate not everyone knows all the intricacies and options with plotting but don't want to have to document or explain absolutely every possible option when the documentation for the various packages is out there and available. From memory I included links to tutorials and documentation for Matplotlib, Pandas and Seaborn to signpost people to these resources.

Reply:
@I see your point on not overcomplicating things, so do feel free to keep it as it is. But just to illustrate the difference this makes, this is an example of bars being not aligned when it's set to `bins=20`
![image](https://user-images.githubusercontent.com/58240320/218136893-b82f9070-4204-4b16-9ba3-75d0496bd270.png)
And with `bins=np.linspace(0, 3e-7, 20)`, the plot now looks like this:
![image](https://user-images.githubusercontent.com/58240320/218136645-d1044c02-51e1-40a9-8543-68ae401da9ea.png)

Reply:
Thanks that is a great example and not something I've ever noticed before. It may be an artifact of Pandas and that using Matplotlib directly doesn't result in this.

Either way I will add this to #462.

Reply:
Realised the above example has not just different ranges, but different frequencies between the groups. I've added a commented line with a note to that effect.